llvm/tools/sycl-post-link/ModuleSplitter.cpp

```diff
@@ -891,5 +930,42 @@ getDeviceCodeSplitter(ModuleDesc &&MD, IRSplitMode Mode, bool IROutputOnly,
   return std::make_unique<ModuleCopier>(std::move(MD), std::move(Groups));
 }
 
+SmallVector<ModuleDesc, 2> splitByESIMD(ModuleDesc &&MD,
+                                        bool EmitOnlyKernelsAsEntryPoints) {
+
+  SmallVector<module_split::ModuleDesc, 2> Result;
+
+  EntryPointGroupVec EntryPointGroups =
+      groupEntryPointsByKernelType(MD, EmitOnlyKernelsAsEntryPoints);
+
+  if (EntryPointGroups.size() == 1) {
+    Result.emplace_back(std::move(MD.releaseModulePtr()),
+                        std::move(EntryPointGroups[0]), MD.Props);
+    return Result;
+  }
+
+  DependencyGraph CG(MD.getModule());
+  for (auto &Group : EntryPointGroups) {
+    if (Group.isEsimd()) {
+      // For ESIMD module, we use full call graph of all entry points and all
+      // ESIMD functions.
+      Result.emplace_back(
+          std::move(extractESIMDCallGraph(MD, std::move(Group), CG)));
```

Comment:
Rename: extractESIMDSubModule?  Thanks

Reply:
Yeah, suggested name better conveys the idea of the function. Renamed in f41d34bb34859749ff52da5a5d70817047c0e649